### PR TITLE
 Inheritance/Abstract - Workers Task - completed

### PR DESCRIPTION
**Pull request by mariastephan2186**
Workers Task Complete

### CODE REVIEW

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+public abstract class Workers {
+    private String name;
+    private String Department;
+
+    //Parent Class Constructor
+    public Workers(String name, String department) {
+        this.name = name;
+        Department = department;
```
Comment (owynowyn) [inline src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java:11]: Would then update this to `this.department = department`

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    public abstract void work();
+
+    //Common method that can be shared by subclasses - Displays Workers basic information.
+
+    public abstract void takeBreak();
+
+    public void displayBasicInfo() {
+        System.out.println("Worker name - " + getName() + " ,Department- " + getDepartment());
```
Comment (owynowyn) [inline src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java:32]: Could just directly call `name` and `department` here as they are local to this class anyway

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/RemoteWorker.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    // Method defining the type of work and the department of the Remote Worker
+    @Override
+    public void work() {
+
+        System.out.println(getName() + " is writing some code during their WFH and belongs to " + getDepartment() + " department");
+
+    }
+// Method that prints the place/duration of the workers' break
+
```
Comment (owynowyn) [inline src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/RemoteWorker.java:19]: Not necessary but adding the @Override annotation here would really stipulate that this is overriding the parent class's method

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/OfficeWorker.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    // Method defining the type of work and the department of the Office Worker
+
+    @Override
+    public void work() {
+
+        System.out.println(getName() + " is processing some docs for the " + getDepartment() + " department");
+
+    }
+// Method that prints the place/duration of the workers' break
+
```
Comment (owynowyn) [inline src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/OfficeWorker.java:21]: Not necessary but adding the @Override annotation here would really stipulate that this is overriding the parent class's method

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/FactoryWorker.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    // Method defining the type of work and the department of the Factory Worker
+    @Override
+    public void work() {
+
+        System.out.println(getName() + " operates on the belt area in the factory and works in the " +
+                getDepartment()+ " department");
+
+    }
+// Method that prints the place/duration of the workers' break
```
Comment (owynowyn) [inline src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/FactoryWorker.java:18]: Not necessary but adding the @Override annotation here would really stipulate that this is overriding the parent class's method

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/OfficeWorker.java (hunk 4 above)
Comment (mariastephan2186) [reply src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/OfficeWorker.java:20]: Added !

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/RemoteWorker.java (hunk 3 above)
Comment (mariastephan2186) [reply src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/RemoteWorker.java:18]: Added !

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java (hunk 2 above)
Comment (mariastephan2186) [reply src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java]: Changed this!

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java (hunk 1 above)
Comment (mariastephan2186) [reply src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java]: Updated this,

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package session9inheritanceabstract.taskentries.WorkersTaskMaria;
+
+
+public abstract class Workers {
+    private String name;
+    private String Department;
```
[the remark this answers is not in the document]
Comment (mariastephan2186) [reply src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/Workers.java]: Changed !

File: src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/FactoryWorker.java (hunk 5 above)
Comment (mariastephan2186) [reply src/main/java/session9inheritanceabstract/taskentries/WorkersTaskMaria/FactoryWorker.java]: @owynowyn Ive made the changes based on comments now!

**Review by owynowyn — APPROVED** [review]
Great work Maria!

**Pull request merged**